src/OpenColorIO/ops/fixedfunction/FixedFunctionOpCPU.cpp

```diff
@@ -653,6 +694,151 @@ void Renderer_ACES_DarkToDim10_Fwd::apply(const void * inImg, void * outImg, lon
     }
 }
 
+Renderer_ACES_GamutMap13_Fwd::Renderer_ACES_GamutMap13_Fwd(ConstFixedFunctionOpDataRcPtr & data)
+    :   OpCPU()
+{
+    m_limCyan        = (float)data->getParams()[0];
+    m_limMagenta     = (float)data->getParams()[1];
+    m_limYellow      = (float)data->getParams()[2];
+    m_thrCyan        = (float)data->getParams()[3];
+    m_thrMagenta     = (float)data->getParams()[4];
+    m_thrYellow      = (float)data->getParams()[5];
+    m_power          = (float)data->getParams()[6];
+
+    m_one_over_power = 1.f / m_power;
+
+    m_thrCyan        = std::min(0.9999f, m_thrCyan);
+    m_thrMagenta     = std::min(0.9999f, m_thrMagenta);
+    m_thrYellow      = std::min(0.9999f, m_thrYellow);
+
+    // Precompute scale factor for y = 1 intersect
+    auto f_scale = [this](float lim, float thr) {
+        return (lim - thr) / powf(powf((1.0f - thr) / (lim - thr), -m_power) - 1.0f, m_one_over_power);
+    };
+    m_scaleCyan      = f_scale(m_limCyan,    m_thrCyan);
+    m_scaleMagenta   = f_scale(m_limMagenta, m_thrMagenta);
+    m_scaleYellow    = f_scale(m_limYellow,  m_thrYellow);
+}
+
+__inline float Renderer_ACES_GamutMap13_Fwd::compress(float dist, float lim, float thr, float scale, bool invert) const
+{
+    float comprDist;
+
+    // No compression below threshold
+    if (dist < thr)
+    {
+        comprDist = dist;
+    }
+    else
+    {
+        // Disable compression, avoid nan
+        if (lim < 1.0001f)
```

Comment:
I moved it a level above in the call stack but it's still in the loop. I didn't found a simple design to move it out the loop completely yet.

Reply:
I guess the question is whether we want to allow lim == 1. (that is, no compression) or whether we simply want to clamp it to 1.0001, which would allow for slightly better performance by moving it out of the pixel loop.  Has there been any guidance from the implementation working group on that topic?  If not, I'm fine with whichever way you prefer.

Reply:
Yes exactly that was my concern about adding a hard clip at 1.0001 in the initialisation phase, tagging @nick for visibility.

Given the time constraint I propose leaving this topic aside and having a follow up performance improvements PR as @hodoulp mentioned. I believe this kind of optimisation is not obviously linked to performance improvements across the board given how good some processors are at branch prediction, this needs through profiling.